File: ISA_FRONTEND/src/app/model/bloodAmount.ts
export class BloodAmount {
    id: string = '';
    bloodType: string = '';
    amount: number = 0;

    public constructor(obj?: any) {
        if (obj) {
            this.id = obj.id;
            this.bloodType = obj.bloodType;
            this.amount = obj.amount;
        }
    }
}

File: ISA_BACKEND/BloodBank/src/main/java/com/example/bloodbank/Model/BloodAmount.java
package com.example.bloodbank.Model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Table(name = "BloodAmount")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class BloodAmount {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private long id;
    private BloodType bloodType;
    private double amount;
    @ManyToOne
    @JoinColumn(name = "blood_bank_id")
    private BloodBank bloodBank;
}
